clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -475,11 +500,20 @@ public Node leastLoadedNode(long now) {
     }
 
     public static AbstractResponse parseResponse(ByteBuffer responseBuffer, RequestHeader requestHeader) {
+        return parseResponseMaybeUpdateThrottleTimeMetrics(responseBuffer, requestHeader, null);
+    }
+
+    private static AbstractResponse parseResponseMaybeUpdateThrottleTimeMetrics(ByteBuffer responseBuffer, RequestHeader requestHeader, Sensor throttleTimeSensor) {
         ResponseHeader responseHeader = ResponseHeader.parse(responseBuffer);
         // Always expect the response version id to be the same as the request version id
         ApiKeys apiKey = ApiKeys.forId(requestHeader.apiKey());
         Struct responseBody = apiKey.parseResponse(requestHeader.apiVersion(), responseBuffer);
         correlate(requestHeader, responseHeader);
+        if (throttleTimeSensor != null) {
+            String throttleTimeFieldName = "throttle_time_ms";
+            if (responseBody.hasField(throttleTimeFieldName))
+                throttleTimeSensor.record(responseBody.getInt(throttleTimeFieldName));
```

Comment:
Most callers are from handleCompletedReceives() which always has the currentTime. Perhaps we could pass that into throttleTimeSensor.record() to avoid an extra call to time.milliseconds().